hypershift-operator/controllers/sharedingress/router.go

```diff
@@ -143,16 +142,21 @@ func ReconcileRouterConfiguration(cm *corev1.ConfigMap, config string) error {
 	return nil
 }
 
-func ReconcileRouterDeployment(deployment *appsv1.Deployment, config *corev1.ConfigMap) error {
+func ReconcileRouterDeployment(deployment *appsv1.Deployment, configMap *corev1.ConfigMap) error {
+	routerDeploymentConfig := config.DeploymentConfig{}
+	routerDeploymentConfig.SetMultizoneSpread(hcpRouterLabels(), false)
+	routerDeploymentConfig.SetNodeSpread(hcpRouterLabels())
```

Comment:
node spread uses a required during scheduling anti affinity would will fail the deployment availability for single node case. Let's use only SetMultizoneSpread for now